# HG changeset patch
# User Rik <rik@octave.org>
# Date 1547012117 28800
#      Tue Jan 08 21:35:17 2019 -0800
# Branch stable
# Node ID 0fba2d918b6543a66da074d59f0b62f147492ed3
# Parent  97a37df42734f2459abdf790fc05e67470f5b60f
pt-eval.cc: Fix static analyzer detected issues (bug #55347).

* pt-eval.cc (tree_evaluator::repl): Free memory in variable "lxr" created by
new.

diff --git a/libinterp/parse-tree/pt-eval.cc b/libinterp/parse-tree/pt-eval.cc
--- a/libinterp/parse-tree/pt-eval.cc
+++ b/libinterp/parse-tree/pt-eval.cc
@@ -185,16 +185,19 @@ namespace octave
     if (retval == EOF)
       {
         if (interactive)
           octave_stdout << "\n";
 
         retval = 0;
       }
 
+    // Clean up memory
+    delete lxr;
+
     return retval;
   }
 
   octave_value_list
   tree_evaluator::eval_string (const std::string& eval_str, bool silent,
                                int& parse_status, int nargout)
   {
     octave_value_list retval;
